### PR TITLE
📁 XDG spec and NOTESH_FILE envvar for default file

### PR DESCRIPTION
**Pull request by nobodyinperson**
This PR introduces a `$NOTESH_FILE` environment variable that controls
the default file Notesh opens. It defaults to `$XDG_DATA_HOME`
(which itself defaults to `~/.local/share`) and can of course still be
overridden by the user via `notesh -f FILE`.

This causes `notesh` to always open the default notes file, no matter
what directory one is in. The user can change the default location by
setting the `$NOTESH_FILE` envvar in their shell config. Setting
`NOTESH_FILE=notes.json` restores the old behaviour.

### CODE REVIEW

File: notesh/main.py [hunk 1]
```diff
@@ -27,12 +28,17 @@ class NoteApp(App):
         Binding("ctrl+c", "quit", "Quit"),
     ]
 
+    DEFAULT_FILE = os.environ.get(
+        "NOTESH_FILE",
+        str(Path(os.environ.get("XDG_DATA_HOME", Path("~/.local/share").expanduser())) / "notesh" / "notes.json"),
+    )
+
```
Comment (Cvaniak) [inline notesh/main.py]: Project currently should work for both Linux and Windows so this part should be universal. I would put this line outside the class and do like:
```python
if os.name == 'nt':  # Windows
    default_dir = Path(os.getenv('APPDATA')) / 'notesh'
else:  # Unix/Linux/Mac
    default_dir = Path(os.getenv("XDG_DATA_HOME", Path("~/.local/share").expanduser())) / 'notesh'

DEFAULT_FILE = Path(os.environ.get("NOTESH_FILE", default_dir / "notes.json"))
```

**Comment by Cvaniak** [comment]
Please apply [black](https://github.com/psf/black) on PR code. There is file [pyproject.toml](https://github.com/Cvaniak/NoteSH/blob/master/pyproject.toml) with config so you can run it from root of the project and it should work.

**Comment by Cvaniak** [comment]
Very good idea and thank you for your contribution!  
For now I had alias for this but your solution seems to be more user friendly.  
I only think that if users want to put notes inside directory they currently are, they will need to write `notes -f notes.json` and maybe we should also add new flag or allow for something like `notes .`.

**Comment by nobodyinperson** [comment]
> Please apply [black](https://github.com/psf/black) on PR code

Didn't run black deliberately because other files would have been touched that haven't been black-formatted. But okay, I'll run black and add the other files as well.

File: notesh/main.py (hunk 1 above)
Comment (nobodyinperson) [reply notesh/main.py]: I kept it inside the class now with an inline if. I think it makes sense to have it attached to the class.

Comment (nobodyinperson) [reply notesh/main.py]: I don't have a Windows box here to test though.

**Pull request merged**